exchanges/kraken/kraken_websocket.go

```diff
@@ -540,17 +542,30 @@ func (k *Kraken) wsProcessTrades(response []any, pair currency.Pair) error {
 		if !ok {
 			return errors.New("unidentified trade data received")
 		}
-		timeData, err := strconv.ParseFloat(t[2].(string), 64)
+		if len(t) < 4 {
+			return fmt.Errorf("unexpected trade data length: %s", t)
+		}
+		ts, ok := t[2].(string)
```

Comment:
TL;DR: This PR doesn't move us forward with cleaner json []any parsing, and that's okay.

OOS Aside (Probably mostly for @shazbert):
Bitfinex managed to put all of this into `wsTrade.UnmarshalJSON` and then put that into trade.Data from there.
We had a note to try that kind of multi-type json slice into references in general.
There's a difference here in that the types don't map cleanly.
I think there's legs in trying to abstract that at some point, and it'd probably require something like types.Float64 or something, because we can't tag decorate json fields in []any.
Closest I can sniff us getting is if we [use a pointer to a type](https://go.dev/play/p/8GU9DmaKIeT)